app/main.py

```diff
@@ -1,3 +1,20 @@
-def cache(func):
-    # Write your code here
-    pass
+from functools import wraps
+from typing import Any, Callable
+
+
+def cache(func: Callable) -> Callable:
+    results = {}
+
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> Any:
+        key = (args, tuple(sorted(kwargs.items())))
+        if key in results:
+            print("Getting from cache")
+            return results[key]
+        else:
+            print("Calculating new result")
+            result = func(*args, **kwargs)
+            results[key] = result
+            return result
```

Comment:
```suggestion
        return results[key]
```

Reply:
You can move return 1 level above, then you'll use just 1 return for both if and else blocks